### ISSUE TITLE
Clarification on Calculating 'Maximum Return' and 'Average Return' Metrics

### ISSUE THREAD

**Issue by sitaharsha** [instadeepai/selective-reincarnation-marl#2]
Hello,

I've been reading your research paper and trying to understand how you calculated the 'maximum return' and 'average return' metrics in your experiments. I have a few questions related to this, and I'm particularly confused about calculating metrics either for timesteps or for episodes.

1. Can you please let me know the specific values of the seeds that you used for running the experiments? I understand that each experiment was repeated over five seeds, but I would like to know the actual seed values used.

2. How did you calculate the 'maximum return' and 'average return' metrics? Did you use a separate script or calculate these directly from the TensorFlow event files? If you used a separate script, can you please share it?

I'd appreciate any help you can provide in understanding these aspects of your work. Thank you in advance for your time and assistance!

**Comment by jcformanek** [comment]
Hi my collaborator worked on this. He is on leave at the moment and will be back on Monday. I will get him to reply as soon as possible.

Also, please note that we only shared a demo dataset in the repo. I will upload the full datasets shortly for download.

**Comment by jcformanek** [comment]
I have uploaded the "Good" and the "Medium" dataset. You can also run experiments on a combination of both datasets, "Good-Medium". I have updated the README with instructions on how to choose the dataset quality. The results in the paper where generated using the "Good-Medium" dataset.

**Comment by sitaharsha** [comment]
Thank you @jcformanek  for uploading the "Good" dataset, and for providing instructions on how to choose the teacher dataset quality. Based on my understanding, the Good dataset consists of time stamps from around 909,000 to 1,109,000, and the Medium dataset ranges from 200,000 to 750,000. So, the Good dataset is taken from the last 20%, and the Medium dataset is from the last 82%. Please correct me if my understanding is not accurate.

**Comment by jcformanek** [comment]
Hi @sitaharsha, thanks for experimenting with our code. I am sorry about the confusion with the timestamps. Because of the way we setup up the offline experience logging at the time, the first few thousand timesteps were not logged correctly. So the file with timestamp zero does not correspond to the first episode during training. We added the footnote about the fraction 20% and 40% as a rough guideline, but its not perfectly accurate. I think its more useful to refer to the histograms of the datasets in the appendix to understand the difference between the two datasets. 

We have fixed the experience logger in the mean time, but the timestamps of the offline data we used for the experiments in the paper is still wrong. In future we will make sure the timestamps are correct. Sorry about that.

**Comment by callumtilbury** [comment]
Hi @sitaharsha! Thanks for checking out our work :)

Re: the maximum and average return metrics, here is the description from the paper:

> For the ‘maximum return’ metric, we find the timestep at which the return, averaged over the five seeds, is highest. For the ‘average return’ metric, we average the return over all seeds and all timesteps. We use these metrics as proxies for performance and speed to convergence respectively.

This calculation was done with a separate script, after all the results had been recorded. Please see a quick sketch below, illustrating the process.

<img width="986" alt="Screenshot 2023-04-17 at 11 01 54" src="https://user-images.githubusercontent.com/37700709/232439242-74d4416a-3844-4190-a635-412d0cb834cc.png">



I trust that provides clarity for you :) If not, I am happy to explain further!

**Comment by sitaharsha** [comment]
Hi @jcformanek and @callumtilbury, thank you for your detailed explanations!

@jcformanek, I appreciate the clarification on the timestamp issue with the offline experience logging. I understand that the first few thousand timesteps were not logged correctly and that the provided fractions are rough guidelines. I'll refer to the histograms in the appendix to better understand the differences between the datasets. It's great to know that the experience logger has been fixed for future experiments.

@callumtilbury, thanks for explaining the maximum and average return metrics and sharing the sketch. It surely helps me understand how the metrics were calculated using a separate script after all the results had been recorded. This information clarifies the process for me.

I'm looking forward to future updates and improvements on this project. Keep up the great work! If I have any further questions or need assistance, I'll reach out. Thanks again!
